internal/cmd/test.go

```diff
@@ -46,10 +72,20 @@ func CMDServerTestingMain(suite *testing.M) {
 		return suite.Run()
 	}()
 
-	os.Exit(code)
+	exit.Exit(code)
 }
 
-func RunCMDTest(test *testing.T, mainFunc func(), args []string) (string, string, int, error) {
+func getUnusedPort() (int, error) {
+	listener, err := net.Listen("tcp", ":0")
```

Comment:
Make sure you some looping tests (just repeatedly calling real fast) on the code that uses this.
(Note that tests looping tests are not something included in your PR, just loke count=100 or the script we made.)
I just want you to make sure that we successfully give up this port that we bound to when checking that it is open.